xarray/core/dataset.py

```diff
@@ -2643,6 +2643,107 @@ def stack(self, dimensions=None, **dimensions_kwargs):
             result = result._stack_once(dims, new_dim)
         return result
 
+    def to_stacked_array(self, new_dim, dims, variable_dim='variable'):
```

Comment:
Maybe we can also use the syntax `stacked = data.to_stacked_array(z=['y'])`

Reply:
While your suggestion is certainly more analogous to `stack`, I think it should be clear that this function only performs a single stacking operation. That is why I chose a more verbose signature.